docs/page/adding-new-language.md

```diff
@@ -11,24 +11,39 @@ corresponding mode -\> language id - in this case `(python-mode .
 
 
 ``` elisp
-(defvar lsp-language-id-configuration
-  '(...
-    (python-mode . "python")
-    ...))
-;; if you are adding the support for your language server in separate repo use
-;; (add-to-list 'lsp-language-id-configuration '(python-mode . "python"))
-
-(lsp-register-client
- (make-lsp-client :new-connection (lsp-stdio-connection "pyls")
-                  :activation-fn (lsp-activate-on "python")
-                  :server-id 'pyls))
+(with-eval-after-load 'lsp-mode
+  (defvar lsp-language-id-configuration
+    '(...
+      (python-mode . "python")
+      ...))
```

Comment:
this should not be in with-eval-after-load. If you are changing the defvar then you should do a PR not put in with-eval-after-load

Reply:
This was in the context of adding an lsp-language-id in one's init.el, where if 'lsp-mode is not yet loaded the variables and functions used become meaningless.

I may have misunderstood the context of this documented example and am happy to remove this change. Is this intended for people contributing to the supported languages?

Reply:
I have reverted the commit that included this change.